src/Modules/UngradedModules/FreeModuleHom.jl

```diff
@@ -511,6 +510,8 @@ function kernel(h::FreeModuleHom{<:FreeMod, <:SubquoModule})
   g = vcat(g, relations(M))
   R = base_ring(G)
   H = FreeMod(R, length(g))
+  # This code is also used by graded modules and we need to care for that.
+  is_graded(h) && is_homogeneous(h) && set_grading!(H, degree.(g))
```

Comment:
Is it possible to add a meaningful test for this? (I assume that the current tests don't really cover this case because they worked before.)